src/utils.rs

```diff
@@ -50,6 +50,58 @@ fn strip_verbatim_from_prefix(prefix: &PrefixComponent<'_>) -> Option<PathBuf> {
     Some(ret)
 }
 
+#[derive(Debug)]
+struct RemoveError {
+    kind: std::io::ErrorKind,
+    path: String,
+}
+
+impl std::error::Error for RemoveError {}
+
+impl std::fmt::Display for RemoveError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!(
+            "failed to remove '{}' : {:?}",
+            self.path, self.kind
+        ))
+    }
+}
+
+pub(crate) fn improve_remove_error(error: std::io::Error, path: &Path) -> std::io::Error {
+    std::io::Error::new(
+        error.kind(),
+        RemoveError {
+            kind: error.kind(),
+            path: path.display().to_string(),
+        },
+    )
+}
+
+#[test]
+fn custom_remove_error() {
+    let path = "test/path".as_ref();
+
+    let expected = "failed to remove 'test/path' : PermissionDenied";
+    let tested = format!(
+        "{}",
+        improve_remove_error(
+            std::io::Error::from(std::io::ErrorKind::PermissionDenied),
+            path
+        )
+    );
+    assert_eq!(expected, tested);
+
+    let expected = "Custom { kind: PermissionDenied, error: RemoveError { kind: PermissionDenied, path: \"test/path\" } }";
```

Comment:
The debug representation is not guaranteed to be stable, and I wouldn't test it.